src/js/profile/profile_loader.js

```diff
@@ -87,8 +95,37 @@ export default class ProfileLoader extends Component {
 
             removePrevCategories = false;
         }
+
+        if (this.watermarkEnabled) {
+            this.addWatermark();
+        }
+    }
+
+    addWatermark() {
+        if ($('.rich-data-nav .watermark-wrapper').length > 0) {
+            return;
+        }
+
+        let watermarkWrapper = document.createElement('div');
+        $(watermarkWrapper)
+            .addClass('watermark-wrapper')
+            .addClass('truncate');
```

Comment:
`.addClass('truncate');`

Oh, that small change is going to make it challenging to separate into a single class. Perhaps it's okay to leave the add watermark logic separate for each element